KenBurns/Classes/KenBurns.swift

```diff
@@ -195,6 +214,53 @@ func ==(lhs: KenBurnsAnimation, rhs: KenBurnsAnimation) -> Bool {
         updatesDisplayLink.add(to: RunLoop.main, forMode: .common)
         startNewAnimation()
     }
+    
+    public func setImageQueue(withImages images:[UIImage]) {
+        if images.count == 0 {
+            fatalError("This is an empty queue!")
+        }
+        
+        remoteQueue = false
+        index = 0
+        self.imageQueue = RingBuffer(count: images.count)
+        
+        for i in images {
+            self.imageQueue?.write(i)
+        }
+    }
+    
+    public func setImageQueue(withUrls urls:[URL], placeholders: [UIImage]?) {
+        guard placeholders == nil || placeholders?.count == urls.count else {
+            fatalError("You don't have a placeholder for every image!")
+        }
+        
+        remoteQueue = true
+        self.imageURLs = RingBuffer<URL>(count: urls.count)
+        self.imagePlaceholders = RingBuffer<UIImage>(count: urls.count)
+        
+        for u in urls {
+            self.imageURLs?.write(u)
+        }
+        
+        self.fetchImage(self.imageURLs!.read()!, placeholder: nil)
+
+        for _ in 0..<10 {
```

Comment:
Not sure I understand where this magic number. 10 is coming from. Can you please explain, and also drop a comment here?

Reply:
From my testing, I found that when the nextImage function is used there is a glitch that remote images might not be downloaded in time. 10 is an arbitrary number.  Of course there are optimizations we could do to avoid this such as image compression, preloading, caching, etc. This is more of a temporary solution